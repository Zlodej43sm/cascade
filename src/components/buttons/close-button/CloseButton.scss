@use '../../../styles/colors';
@use '../../../styles/mixins';
@use '../../../styles/sizes';

.button {
  @include mixins.resetButtonStyles;
  @include mixins.crossMark(sizes.$icon-md, 2px, colors.$base-grey);
  width: sizes.$button-sm;
  height: sizes.$button-sm;
  border-radius: 50%;
  padding: 0;
}
