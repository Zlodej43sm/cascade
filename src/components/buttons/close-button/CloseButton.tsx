import React from 'react';

import styles from './CloseButton.scss';

export type CloseButtonProps = {
  onClose?: (event: React.MouseEvent<HTMLButtonElement>) => void
};

function CloseButton({ onClose }: CloseButtonProps) {
  return (
    <button
      className={styles.button}
      onClick={onClose}
      type="button"
      aria-label="Close"
      data-testid="close-button"
    />
  );
}

export default CloseButton;
