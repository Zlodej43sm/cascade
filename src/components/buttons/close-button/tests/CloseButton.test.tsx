import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';

import { closeButtonArguments } from '../__mocks__/CloseButton.mock';
import CloseButton from '../CloseButton';

describe('CloseButton', () => {
  it('Should match snapshot', () => {
    const { asFragment } = render(<CloseButton {...closeButtonArguments} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('Should call close handler', () => {
    const onClose = jest.fn();
    render(<CloseButton onClose={onClose} />);
    fireEvent.click(screen.getByTestId('close-button'));
    expect(onClose).toHaveBeenCalled();
  });
});
