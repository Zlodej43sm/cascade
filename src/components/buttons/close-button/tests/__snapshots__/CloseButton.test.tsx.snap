// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseButton Should match snapshot 1`] = `
<DocumentFragment>
  <button
    aria-label="Close"
    class="button"
    data-testid="close-button"
    type="button"
  />
</DocumentFragment>
`;
